drop-cap/library/src/main/java/com/novoda/dropcap/DropCapView.java

```diff
@@ -246,49 +251,61 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         super.onMeasure(widthMeasureSpec, desiredHeightMeasureSpec);
     }
 
-    private void measureDropCapFor(int width) {
-        dropCapWidth = (int) (dropCapPaint.measureText(dropCapText, 0, 1) + spacer);
-        dropCapPaint.getTextBounds(dropCapText, 0, 1, dropCapBounds);
-        int copyWidthForDropCap = width - dropCapWidth;
+    private void measureDropCapFor(int totalWidth) {
+        dropCapWidth = (int) (dropCapPaint.measureText(dropCapText, 0, dropCapText.length()) + spacer);
+        dropCapPaint.getTextBounds(dropCapText, 0, dropCapText.length(), dropCapBounds);
+        int remainingWidthAfterDropCap = totalWidth - dropCapWidth;
 
-        if (dropCapStaticLayout == null || dropCapStaticLayout.getWidth() != copyWidthForDropCap) {
-            dropCapStaticLayout = new StaticLayout(
-                    dropCapText + copyText,
+        if (dropCapCopyStaticLayout == null || dropCapCopyStaticLayout.getWidth() != remainingWidthAfterDropCap) {
+            dropCapCopyStaticLayout = new StaticLayout(
+                    copyText,
                     copyTextPaint,
-                    copyWidthForDropCap,
+                    remainingWidthAfterDropCap,
                     Layout.Alignment.ALIGN_NORMAL,
                     SPACING_MULTIPLIER,
                     lineSpacingExtra,
                     true
             );
 
             calculateLinesToSpan();
+            calculateDropCapBaseline();
 
-            if (enoughLinesForDropCap()) {
-                float baseline = dropCapBounds.height() + getPaddingTop();
-                dropCapBaseline = baseline - dropCapBounds.bottom;
-            }
+            canDrawDropCap = enoughLinesToWrapDropCap() && dropCapCopyFitsWithin(totalWidth);
```

Comment:
The rules determining whether a `DropCap` can be drawn are only present in the `measureDropCapFor` now instead of relying on checks to `enoughLinesToWrapDropCap` all over the codebase.